rotkehlchen/db/updates.py

```diff
@@ -59,7 +55,56 @@ def __init__(self, msg_aggregator: 'MessagesAggregator', user_db: 'DBHandler') -
             UpdateType.RPC_NODES: self.update_rpc_nodes,
             UpdateType.CONTRACTS: self.update_contracts,
             UpdateType.GLOBAL_ADDRESSBOOK: self.update_global_addressbook,
-        }
+        }  # If we ever change this also change tests/unit/test_data_updates::reset_update_type_mappings  # noqa: E501
+        self.version = pversion.parse(get_current_version().our_version)
+
+    def update_single(
+            self,
+            update_type: UpdateType,
+            from_version: int,
+            to_version: int,
+            limits: dict[str, dict],
+    ) -> None:
+        """Common code to update a single type of data
+
+        Check the updates in the inclusive range [from_version + 1, to_version] and apply
+        the needed updates if any. Also save latest applied update type in the DB.
+        """
+        for update_version in range(from_version + 1, to_version + 1):
+            version_info = limits.get(str(update_version), {})  # json treats keys as string
+            min_version = version_info.get('min_version', None)
+            if min_version is not None:
+                p_min_version = pversion.parse(min_version)
+                if p_min_version > self.version:
+                    log.warning(f'Not updating {update_type.value} to {update_version=} due to {min_version=}')  # noqa: E501
+                    break  # stop iteration since all next versions will also hit this. User needs to update  # noqa: E501
+
+            max_version = version_info.get('max_version', None)
+            if max_version is not None:
+                p_max_version = pversion.parse(max_version)
+                if p_max_version < self.version:
+                    log.warning(f'Not updating {update_type.value} to {update_version=} due to {max_version=}')  # noqa: E501
+                    continue  # probably can never apply this update. Check next one
+
+            file_url = f'https://raw.githubusercontent.com/rotki/data/{self.branch}/updates/{update_type.value}/v{update_version}.json'  # noqa: E501
+            try:
+                updates = query_file(file_url, True)
+            except RemoteError as e:
+                log.warning(f'Failed to update {update_type.value} due to {e!s}')
+                continue  # perhaps broken file? Skipping
+
+            updated_data = updates.get(update_type.value)
+            if updated_data is None:
+                log.error(f'Remote update {file_url} does not contain {update_type.value} key')
+                continue  # perhaps broken file? Skipping
```

Comment:
this is the same comment as in the remote error. I guess remote error should be broken path? or is it intended to be the same?

Reply:
broken file is intended to include both as a meaning but can specify if you want. I just wanted to explain why "continue"